primitives/src/v1/mod.rs

```diff
@@ -1012,6 +1045,35 @@ pub struct AbridgedHrmpChannel {
 	pub mqc_head: Option<Hash>,
 }
 
+/// A possible upgrade restriction that prevents a parachain from performing an upgrade.
+#[derive(Encode, Decode, PartialEq, RuntimeDebug)]
+pub enum UpgradeRestriction {
```

Comment:
It would be good to have explicit indices on these enums.

Reply:
https://github.com/paritytech/polkadot/pull/3371/commits/b058f8e6ff600d0a113bf21256f40c7b53b84267